lib/Core/Searcher.cpp

```diff
@@ -104,8 +104,11 @@ void BFSSearcher::update(ExecutionState *current,
   if (!addedStates.empty() && current &&
       std::find(removedStates.begin(), removedStates.end(), current) ==
           removedStates.end()) {
-    assert(states.front() == current);
-    states.pop_front();
+    std::deque<ExecutionState*>::iterator pos = std::find(states.begin(),
```

Comment:
If you can, please replace `std::deque<ExecutionState*>::iterator` with `auto`.

Reply:
No problem!